cooperator/models/partner.py

```diff
@@ -227,6 +229,11 @@ def get_representative(self):
         return self.child_ids.filtered("representative")
 
     def get_cooperator_from_email(self, email):
+        if not email:
+            return self.browse()
+        email = email.strip()
+        if not email:
+            return self.browse()
```

Comment:
good point. this could also be written:
```python
if email:
    email = email.strip()
if not email:
    return self.browse()
```
or comments added to the current version:
```python
# email could be falsy
if not email:
    return self.browse()
email = email.strip()
# or could be only made of whitespace. this test must be done twice
# because None or False have no .strip() method.
if not email:
    return self.browse()
```
which version do you prefer?

Reply:
The second one is more helpful to the reader than the first one, and could prevent a future error.

Reply:
what about this?
```python
# email could be falsy
if email:
    email = email.strip()
# email could be falsy or be only made of whitespace.
if not email:
    return self.browse()
```

Reply:
yes, even better :)

Reply:
actually I still prefer the second one because it is easier to understand, but maybe that's just me ^^

Reply:
@robinkeunen?

Reply:
The version with the fewest returns I like best.